src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1842,7 +1843,14 @@ int uct_dc_mlx5_ep_is_connected(const uct_ep_h tl_ep,
         rgid = NULL;
     }
 
-    return uct_ib_iface_is_same_device(
-            (const uct_ib_address_t*)params->device_addr, ntohs(ep->av.rlid),
-            rgid);
+    dct = ep->av.dqp_dct;
+    if (!iface->ud_common.config.compact_av ||
+        (ep->flags & UCT_DC_MLX5_EP_FLAG_GRH)) {
+        dct &= ~UCT_IB_MLX5_EXTENDED_UD_AV;
```

Comment:
use UCS_MASK(UCT_IB_QPN_ORDER) - after converting to host endianess
no need to check for compact_av explicitly